### PR TITLE
update usage to have links to Composer and Frontend

### PR DESCRIPTION
**Pull request by akash1810**
Previously, we were just linking to Composer. Now we have icons linking to Frontend and Composer.

Also, show creation date relative to now.

image

Before Deploying
- [x] update configuration with url to Viewer

### CODE REVIEW

**Comment by akash1810** [comment]
cc @mr-mr

File: public/video-ui/src/components/Icon.js [hunk 1]
```diff
@@ -1,5 +1,29 @@
 import React from 'react';
 
+export class FrontendIcon extends React.Component {
+  render () {
+    return (
+      <span className="icon svg-icon">
+        <svg xmlns="http://www.w3.org/2000/svg" width="128" height="128" viewBox="0 0 128 128">
```
Comment (akash1810) [inline public/video-ui/src/components/Icon.js]: stolen from https://github.com/guardian/grid/tree/master/kahuna/public/js/components/gr-icon/icons 😉

File: public/video-ui/src/actions/VideoActions/getVideos.js [hunk 2]
```diff
@@ -66,12 +66,10 @@ function adaptCapiAtom(atom) {
 }
 
 export function searchVideosWithQuery(query) {
-  const encodedQuery = encodeURIComponent(query);
```
Comment (akash1810) [inline public/video-ui/src/actions/VideoActions/getVideos.js]: this has been moved into `capi.js`, closer to where its needed

**Comment by jennysivapalan** [comment]
Should we link to viewer (preview) too? Then the user can check mobile and apps too if they wish. Though

**Comment by akash1810** [comment]
@jennysivapalan great idea! Will add it.

**Comment by akash1810** [comment]
> Should we link to viewer (preview) too? 

Added.

image

File: public/video-ui/src/components/Icon.js [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+      </span>
+    );
+  }
+}
+
+export class ViewerIcon extends React.Component {
+  render () {
+    return (
+      <span className="icon svg-icon">
+        <svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
```
Comment (akash1810) [inline public/video-ui/src/components/Icon.js]: copied from https://github.com/guardian/editorial-viewer/blob/master/public/images/eye-white-32.svg

**Comment by akash1810** [comment]
This breaks the `create new page` button... blocking merging for now.

**Comment by akash1810** [comment]
> This breaks the create new page button

Fixed by polling CAPI up to 10 seconds because it takes a little time for new Composer pages to reach it.

polling

File: public/video-ui/src/components/VideoUsages/VideoUsages.js [hunk 4]
```diff
@@ -71,11 +73,28 @@ export default class VideoUsages extends React.Component {
   };
 
   renderUsage = (usage) => {
-    const composerLink = `${this.getComposerUrl()}/find-by-path/${usage}`;
+    const composerLink = `${this.getComposerUrl()}/content/${usage.fields.internalComposerCode}`;
+    const viewerLink = `${this.getViewerUrl()}/preview/${usage.id}`;
+    const websiteLink = `https://gu.com/${usage.id}`;
```
Comment (jennysivapalan) [inline public/video-ui/src/components/VideoUsages/VideoUsages.js]: I would use `https://www.guardian.com` and save a redirect!

Comment (akash1810) [inline public/video-ui/src/components/VideoUsages/VideoUsages.js]: ah, yep fair point. I was being lazy 😄 . Updated.

File: public/video-ui/styles/abstracts/_variables.scss [hunk 5]
```diff
@@ -89,3 +89,4 @@ $anchorColor: $cWhite;
 $toolbarHeight: 50px;
 $videoBlockWidth: 230px;
 $standard-padding: 10px;
+$iconSize: 16px; /* Preferred icon size */
```
Comment (mbarton) [inline public/video-ui/styles/abstracts/_variables.scss]: 👍 I was looking for this in my branch

**Comment by akash1810** [comment]
> I'm on the fence about polling CAPI rather than updating state locally immediately then merging it with the real state once updated.
I guess in reality if the page doesn't appear in CAPI, that will appear as a problem for the user (won't render in preview) so it's worth the poll.

`VideoUsage` renders properties from a capi response (e.g. `usage.fields.internalComposerCode`). As the capi object is complex, It looked a bit messy to have a create and use a fake capi object then replace it with the real capi object.

Polling seemed like a lesser evil.

**Pull request merged**